RELEASE.md

```diff
@@ -1,13 +1,4 @@
-# Upcoming release:
```

Comment:
This section should be kept as this serves as a great checklist for drafting the release notes, moreover it is not displayed on the github release notes, so no problem in keeping that?

Reply:
I thought we would add it back after the release?